Assets/__Scripts/MapEditor/MapExporter.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using QuestDumper;
+using UnityEngine.Localization.Settings;
+using Debug = UnityEngine.Debug;
+using Task = System.Threading.Tasks.Task;
+
+// This is a struct because it doesn't really manage data
+// Should this just be a static singleton since `song` is always passed from BeatSaberSongContainer?
+public struct MapExporter
+{
+    // TODO: Move constants
+    public const string QUEST_CUSTOM_SONGS_LOCATION =
+        "sdcard/ModData/com.beatgames.beatsaber/Mods/SongLoader/CustomLevels";
+
+    // I added this so the non-quest maintainers can use it as a reference for adding WIP uploads
+    // this does indeed exist and work, please don't refrain from asking me. 
+    public const string QUEST_CUSTOM_SONGS_WIP_LOCATION =
+        "sdcard/ModData/com.beatgames.beatsaber/Mods/SongLoader/CustomWIPLevels";
+
+
+    private readonly BeatSaberSong song;
+
+    public MapExporter(BeatSaberSong song) => this.song = song;
+
+    /// <summary>
+    /// Exports the files to the Quest using adb
+    /// </summary>
+    public async Task ExportToQuest()
+    {
+        var (devices, output) = await Adb.GetDevices();
```

Comment:
If the user does the `OnQuestSave` keybind, we don't check if Adb is installed before here so an `InvalidOperationException` can happen.